### PR TITLE
[GOAL2-614] Updated algoh to capture algod logs on early termination

### PR DESCRIPTION
**Pull request by egieseke**
Please review this change to capture logs In case algod terminates before block watcher initializes.

### CODE REVIEW

File: cmd/algoh/main.go [hunk 1]
```diff
@@ -60,12 +60,13 @@ func (c *stdCollector) Write(p []byte) (n int, err error) {
...
 
 func main() {
+	blockWatcherInitialized := false
 	flag.Parse()
 	nc := getNodeController()
 
 	genesisID, err := nc.GetGenesisID()
 	if err != nil {
-		fmt.Fprintln(os.Stdout, "error loading telemetry config", err)
+		_, _ = fmt.Fprintln(os.Stdout, "error loading telemetry config", err)
```
Comment (Karmastic) [inline cmd/algoh/main.go]: What's the purpose of the "_, _"?  Were you getting lint errors or something?

File: cmd/algoh/main.go [hunk 2]
```diff
@@ -126,11 +127,32 @@ func main() {
...
+		err = cmd.Wait()
+		if err != nil {
+			reportErrorf("error waiting for algod: %v", err)
 		}
-		cmd.Wait()
 		close(done)
 
+		// capture logs if algod terminated prior to blockWatcher starting
+		if !blockWatcherInitialized && algohConfig.UploadOnError {
+			if errorOutput.output != "" {
```
Comment (Karmastic) [inline cmd/algoh/main.go]: This looks like a duplicate of the code below - please consolidate.  Also - this block should not all be gated by 'UploadOnError' - that should only affect whether we call sendLogs(), like below.

File: cmd/algoh/main.go (hunk 1 above)
Comment (egieseke) [reply cmd/algoh/main.go]: OK, Goland required to get the green check for code analysis.  I will remove.

File: cmd/algoh/main.go (hunk 2 above)
Comment (egieseke) [reply cmd/algoh/main.go]: OK,Refactored into a separate method collectErrorLogs().  Removed UploadOnError condition.

**Pull request merged**